gcp/modules/agentless-scanner-service-account/main.tf

```diff
@@ -10,6 +10,8 @@ locals {
   project_id = data.google_client_config.current.project
   # Use provided unique_suffix or generate random one
   effective_suffix = var.unique_suffix != "" ? var.unique_suffix : random_id.deployment_suffix[0].hex
+  # Extract secret name from full path (projects/PROJECT_ID/secrets/SECRET_NAME -> SECRET_NAME)
+  secret_name = regex("projects/[^/]+/secrets/(.+)$", var.api_key_secret_id)[0]
```

Comment:
```tf
"^projects/[^/]+/secrets/(.+)$"
```

Reply:
we could probably rely on the same regexp as our validation regexp in our instance module (variables.tf), just to make maintaining this a little easier.

`var.api_key_secret_id == null || can(regex("^projects/[a-zA-Z0-9-]+/secrets/[a-zA-Z0-9-]+$", var.api_key_secret_id))`